bundles/org.openhab.binding.meteofrance/src/main/java/org/openhab/binding/meteofrance/internal/handler/RainForecastHandler.java

```diff
@@ -0,0 +1,155 @@
+
+        ForecastConfiguration config = getConfigAs(ForecastConfiguration.class);
+        location = Optional.of(new PointType(config.location));
+
+        refreshJob = Optional.of(scheduler.schedule(this::updateAndPublish, 2, TimeUnit.SECONDS));
+    }
+
+    private void disposeJob() {
+        refreshJob.ifPresent(job -> job.cancel(true));
+        refreshJob = Optional.empty();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Rain Forecast handler.");
+        disposeJob();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAndPublish();
+        }
+    }
+
+    private void updateAndPublish() {
+        getBridgeHandler().ifPresentOrElse(handler -> {
+            location.ifPresent(loc -> {
+                RainForecast forecast = handler.getRainForecast(loc);
+                if (forecast != null) {
+                    setProperties(forecast.properties);
+                    updateDate(UPDATE_TIME, forecast.updateTime);
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            });
+        }, () -> logger.warn("No viable bridge"));
+    }
+
+    private void setProperties(@Nullable Properties forecastProps) {
+        if (forecastProps == null) {
+            return;
+        }
+        Map<String, String> properties = editProperties();
+        properties.put("altitude", "%d m".formatted(forecastProps.altitude));
+        properties.put("name", forecastProps.name);
+        properties.put("country", forecastProps.country);
+        properties.put("department", forecastProps.frenchDepartment);
+        properties.put("timezone", forecastProps.timezone);
+        properties.put("comfidence", "%d".formatted(forecastProps.confidence));
+        this.updateProperties(properties);
+
+        setForecast(forecastProps.forecast);
+    }
+
+    private void setForecast(List<Forecast> forecast) {
+        TimeSeries timeSeries = new TimeSeries(REPLACE);
+
+        forecast.forEach(prevision -> {
+            State state = prevision.rainIntensity() != RainIntensity.UNKNOWN
+                    ? new DecimalType(prevision.rainIntensity().ordinal())
+                    : UnDefType.UNDEF;
+            timeSeries.add(prevision.time().toInstant(), state);
+        });
+        sendTimeSeries(intensityChannelUID, timeSeries);
+
+        Forecast current = forecast.get(0);
+        long until = ZonedDateTime.now().until(current.time(), ChronoUnit.SECONDS);
+        logger.debug("Refresh rain intensity forecast in : {}s", until);
+        refreshJob = Optional.of(scheduler.schedule(this::updateAndPublish, until, TimeUnit.SECONDS));
+    }
+
+    private void updateDate(String channelId, @Nullable ZonedDateTime zonedDateTime) {
+        if (isLinked(channelId)) {
+            updateState(channelId, zonedDateTime != null ? new DateTimeType(zonedDateTime) : UnDefType.NULL);
```

Comment:
AFAIK the framework does not automatically change the zoneddatetime to the configured timezone of openHAB.
Not sure in what timezone the API responds, but it would be nice to use the openHAB api -> `timezoneprovider` to update it to the openHAB configured timezone.

Reply:
You're more than right.